docs/docs/20-concepts/02-mcp-gateway/03-obot-registry.md

```diff
@@ -0,0 +1,288 @@
+- The MCP endpoint must be available at the specified path
+- Image must be publicly accessible (Docker Hub, GHCR, etc.)
+
+---
+</details>
+
+
+<details>
+  <summary>Option C: Stdio MCP Server (npx, uvx, docker)</summary>
+
+**⚠️ Important:** If your server is stdio-based, do NOT complete this section yet.
+**Instead:**
+
+1. Complete **Step 2** below to repackage your stdio server
+2. Wait for your image to be published to GHCR
+3. Return here and use this runtime block:
+
+```yaml
+runtime: containerized
+containerizedConfig:
+  image: ghcr.io/obot-platform/mcp-images/<your-server-name>:<tag>
+  port: 8099         # Fixed for stdio servers
+  path: /            # Fixed for stdio servers
+  args:
+    - <your-mcp-server-command>           # Command to run your stdio server
+    # Add any additional flags your server needs:
+    # - --region
+    # - us-east-1
+
+```
+
+The `args` must include the command that starts your stdio server, plus any required flags.
+
+---
+</details>
+
+---
+
+---
+
+## Step 2 — (Only for stdio) Repackage and publish an image
+
+If your server is stdio-based and you want Obot to run it, we’ll build and publish a helper image.
+
+Fork from: `https://github.com/obot-platform/mcp-images`
+
+Edit `repackaging/images.yaml` and add your entry:
+
+### Node.js (npx) Servers
+
+```yaml
+- name: my-awesome-server
+  type: node
+  package: @myorg/mcp-awesome-server    # NPM package name
+  version: 1.2.3                        # Specific version
+
+```
+
+### Python (uvx) Servers
+
+```yaml
+- name: my-python-server
+  type: python
+  package: mcp-python-server            # PyPI package name
+  version: 2.0.1
+
+```
+
+### After Your Image is Published
+
+Once your PR in `mcp-images` merges, we'll publish:
+
+```
+ghcr.io/obot-platform/mcp-images/<your-server-name>:<version>
+
+```
+
+**Then** return to Step 1 and complete Option C with your published GHCR image.
+
+---
+
+## Step 3 — Open your PR(s)
+
+### For Remote or Containerized HTTP Servers
+
+Open **one PR** in `mcp-catalog` adding `your-server-name.yaml`
+
+### For Stdio Servers
+
+1. Open a PR in `mcp-images` editing `repackaging/images.yaml`
+2. Wait for the image to be published to GHCR (typically within a few hours after the PR is merged).
+3. Open a PR in `mcp-catalog` referencing the published GHCR image
+
+---
+
+## Pre-Submission Checklist
+
+Before opening your PR, verify:
+
+- [ ]  `description` includes **`## Features`** and **`## What you'll need to connect`** sections
+- [ ]  `metadata.categories` lists a few categories
+- [ ]  `icon` is a square image, publicly accessible
+- [ ]  `repoURL` points to GitHub (or documentation if unavailable)
+- [ ]  `toolPreview` lists your tool schemas (top 20 if you have many)
+- [ ]  **Exactly one** runtime block (remote OR containerized)
+- [ ]  All `env` variables have clear descriptions
+- [ ]  Sensitive credentials are marked `sensitive: true`
+- [ ]  (Stdio only) Image is published to GHCR before catalog PR
```

Comment:
You should double check all references to this idea - we are ALWAYS repackaging mcp servers except when they are fully remote